osa/scripts/gain_selection.py

```diff
@@ -123,27 +125,47 @@ def apply_gain_selection(date: str, output_basedir: Path = None):
         
         for file in r0_files:
             output_file = output_dir / file.name
-            file.link_to(output_file)
+            sp.run(["cp", file, output_dir])
+
+def run_sacct_j(job) -> StringIO:
+    """Run sacct to obtain the job information."""
+    if shutil.which("sacct") is None:
+        log.warning("No job info available since sacct command is not available")
+        return StringIO()
+
+    sacct_cmd = [
+        "sacct",
+        "-n",
+        "--parsable2",
+        "--delimiter=,",
+        "--units=G",
+        "-o",
+        ",".join(FORMAT_SLURM),
+        "-j",
+        job,
+    ]
+    
+    return StringIO(sp.check_output(sacct_cmd).decode())
 
 def check_failed_jobs(date: str, output_basedir: Path = None):
     """Search for failed jobs in the log directory."""
     failed_jobs = []
-    log_dir = output_basedir / "log" /date
-    filenames = log_dir.glob("gain_selection*.log")
+    log_dir = output_basedir / "log" / date
+    filenames = glob.glob(f"{log_dir}/gain_selection*.log")
+    jobs = [re.search(r'(?<=_)(.[0-9.]+?)(?=.log)', i).group(0) for i in filenames]
 
-    for line in fileinput.input(filenames):
-        if re.search("FAILED", line) or re.search("Stream [1-4] not found", line):
-            job_id = str(fileinput.filename())[-12:-4]
-            run_id = str(fileinput.filename())[-23:-18]
-            subrun_id = str(fileinput.filename())[-17:-13]
-            failed_jobs.append(job_id)
-
-            log.warning(f"Job {job_id} (corresponding to run {run_id}, subrun {subrun_id}) failed.")
+    for job in jobs: 
```

Comment:
I do not know if it's better to loop over the jobs or simply make a single sacct call based on time. However, you would then need to filter the rows corresponding to `jobs`. How long does it take to run it with the loop over jobs?

Reply:
It takes approximately 3 minutes per night... I will check if it's faster doing it with the `starttime` option

Reply:
Probably not. Do not bother.